crates/telio-dns/src/nameserver.rs

```diff
@@ -253,13 +259,30 @@ impl LocalNameServer {
             return Err(String::from("Invalid IpPacket"));
         }
 
+        let payload = match ip.next_level_protocol() {
+            IpNextHeaderProtocols::Udp => LocalNameServer::process_udp_packet(&ip)?,
+            IpNextHeaderProtocols::Tcp => {
+                let tcp_request = TcpPacket::new(ip.payload())
+                    .ok_or_else(|| String::from("Failed to build TcpPacket from request packet"))?;
+                PayloadRequestInfo::Tcp {
+                    source_port: tcp_request.get_source(),
+                    destination_port: tcp_request.get_destination(),
+                }
+            }
+            _ => {
+                return Err(String::from("Invalid protocol for DNS request"));
+            }
+        };
+
         Ok(RequestInfo {
             ip: ip.ip_request_info(),
-            udp: LocalNameServer::process_udp_packet(&ip)?,
+            payload,
         })
     }
 
-    fn process_udp_packet<'a>(ip_request: &impl IpPacket<'a>) -> Result<UdpRequestInfo, String> {
+    fn process_udp_packet<'a>(
```

Comment:
This is a kind of a nitpick, so treat it with low-prio ;). But maybe it makes sense to make `process_udp_packet()` strong-typed, and pass `UdpPacket` as an argument instead of `IpPacket` as an argument?

Reply:
fixed